### ISSUE TITLE
hashCode and equals underwriting for interface-defined value types

### ISSUE THREAD

**Issue by elucash** [immutables/immutables#739]
interface-defined value types are broadly used in a lot of codebases, yet there's no way to specify manually written customized `hashCode` and `equals` implementations in the same way it can be done for abstract classes: default methods are not allowed to redefine methods from `java.lang.Object`.

**Comment by foal** [comment]
And toString too.

**Comment by elucash** [comment]
(WIP) Example usage

```java
@Value.Style(underrideEquals = "equalTo", underrideHashCode = "hash", underrideToString = "stringify")
@Value.Immutable
public interface UnderrideObjectMethods {
  int a();

  default String stringify() {
    return "%%%";
  }
  default int hash() {
    return -1;
  }
  default boolean equalTo(UnderrideObjectMethods b) {
    return this == b;
  }

  @Value.Style(underrideEquals = "equalTo", underrideHashCode = "hash", underrideToString = "stringify")
  @Value.Immutable
  interface StaticUnderride {
    static String stringify(StaticUnderride a) {
      return "!!!";
    }
    static int hash(StaticUnderride a) {
      return -2;
    }
    static boolean equalTo(StaticUnderride a, StaticUnderride b) {
      return a == b;
    }
  }
}
```

**Comment by elucash** [comment]
released in 2.9.0
